android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java

```diff
@@ -2358,6 +2372,53 @@ public static void onMethodCall(MethodCall call, MethodChannel.Result result, Vi
 
     // Methods
 
+    private static void groupAnnotations(MethodCall call, MethodChannel.Result result, ViewerComponent component) throws PDFNetException, JSONException {
+        String annotation = call.argument(KEY_ANNOTATION);
+        String annotationList = call.argument(KEY_ANNOTATION_LIST);
+        PDFViewCtrl pdfViewCtrl = component.getPdfViewCtrl();
+        PDFDoc pdfDoc = component.getPdfDoc();
+
+        if (null == pdfViewCtrl || null == pdfDoc) {
+            result.error("InvalidState", "Activity not attached", null);
+            return;
+        }
+
+        boolean shouldUnlock = false;
+        try {
+            pdfViewCtrl.docLock(true);
+            shouldUnlock = true;
+
+            if (null != annotationList && annotation != null) {
+
+                JSONObject annotationJson = new JSONObject(annotation);
+                String annotationId = annotationJson.getString(KEY_ANNOTATION_ID);
+                int annotationPageNumber = annotationJson.getInt(KEY_PAGE_NUMBER);
+                Annot primaryAnnotation = ViewerUtils.getAnnotById(pdfViewCtrl, annotationId, annotationPageNumber);
+
+                JSONArray annotationJsonArray = new JSONArray(annotationList);
+                ArrayList<Annot> subAnnotations = new ArrayList<>(annotationJsonArray.length());
+                for (int i = 0; i < annotationJsonArray.length(); i++) {
+                    JSONObject currAnnot = annotationJsonArray.getJSONObject(i);
+                    if (currAnnot != null) {
+                        String currAnnotId = currAnnot.getString(KEY_ANNOTATION_ID);
+                        int currAnnotPageNumber = currAnnot.getInt(KEY_PAGE_NUMBER);
+                        if (!Utils.isNullOrEmpty(currAnnotId)) {
+                            Annot validAnnotation = ViewerUtils.getAnnotById(pdfViewCtrl, currAnnotId, currAnnotPageNumber);
+                            if (validAnnotation != null && validAnnotation.isValid()) {
+                                subAnnotations.add(validAnnotation);
+                            }
+                        }
+                    }
+                }
+                AnnotUtils.createAnnotationGroup(pdfViewCtrl, primaryAnnotation, subAnnotations);
```

Comment:
is subAnnotations here all annot in group? if you read android API the third param is all in group @eamon-mallon